x/ccv/provider/keeper/key_assignment.go

```diff
@@ -0,0 +1,739 @@
+package keeper
+
+import (
+	"errors"
+	"sort"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	providertypes "github.com/cosmos/interchain-security/x/ccv/provider/types"
+
+	sdkcryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	ccv "github.com/cosmos/interchain-security/x/ccv/types"
+	tmprotocrypto "github.com/tendermint/tendermint/proto/tendermint/crypto"
+)
+
+type VSCID = uint64
+type ProviderPublicKey = tmprotocrypto.PublicKey
+type ConsumerPublicKey = tmprotocrypto.PublicKey
+type ProviderConsAddr = sdk.ConsAddress
+type ConsumerConsAddr = sdk.ConsAddress
+
+func TMCryptoPublicKeyToConsAddr(k tmprotocrypto.PublicKey) sdk.ConsAddress {
+	sdkK, err := sdkcryptocodec.FromTmProtoPublicKey(k)
+	if err != nil {
+		panic("could not get public key from tm proto public key")
+	}
+	return sdk.GetConsAddress(sdkK)
+}
+
+type Store interface {
+	SetProviderConsAddrToConsumerPublicKey(ProviderConsAddr, ConsumerPublicKey)
+	GetProviderConsAddrToConsumerPublicKey(ProviderConsAddr) (ConsumerPublicKey, bool)
+	DelProviderConsAddrToConsumerPublicKey(ProviderConsAddr)
+	IterateProviderConsAddrToConsumerPublicKey(func(ProviderConsAddr, ConsumerPublicKey) bool)
+	SetConsumerPublicKeyToProviderPublicKey(ConsumerPublicKey, ProviderPublicKey)
+	GetConsumerPublicKeyToProviderPublicKey(ConsumerPublicKey) (ProviderPublicKey, bool)
+	DelConsumerPublicKeyToProviderPublicKey(ConsumerPublicKey)
+	IterateConsumerConsAddrToLastUpdateMemo(func(ConsumerConsAddr, providertypes.LastUpdateMemo) bool)
+	SetConsumerConsAddrToLastUpdateMemo(ConsumerConsAddr, providertypes.LastUpdateMemo)
```

Comment:
imo might be useful to add VSC into here instead of just "update", or put it in comments

Reply:
Also removing "memo" as we talked about before, I'm sure that'd touch a lot of places in code tho